// client/src/pages/Home.jsx
function ProductCard({ p, i }) {
    return (
        <div
            className="bg-white rounded-2xl overflow-hidden border border-gray-100 card-hover slide-in"
            style={{ animationDelay: `${i * 0.1}s` }}
        >
            <div className="h-48 bg-gradient-to-br from-sun-100 to-ocean-50 flex items-center justify-center text-6xl">
                {p.emoji}
            </div>

            <div className="p-5">
                <p className="text-xs text-sun-600 font-semibold mb-1">
                    {p.category}
                </p>

                <h3 className="font-bold text-lg mb-1">
                    {p.name}
                </h3>

                <div className="flex items-center gap-2 mb-3">
                    <span className="text-yellow-500 text-sm">
                        ★ {p.rating}
                    </span>

                    <span className="text-gray-400 text-xs">
                        | {p.brand}
                    </span>
                </div>

                <div className="flex items-center justify-between">
                    <span className="text-xl font-bold text-sun-600">
                        ${p.price}
                    </span>

                    <button className="px-4 py-2 bg-sun-500 text-white text-sm font-medium rounded-xl hover:bg-sun-600 transition">
                        View Details
                    </button>
                </div>
            </div>
        </div>
    );
}


const products = [
    {
        id: 1,
        name: "UV Protection Sunglasses",
        brand: "SunShade",
        price: 15,
        rating: 4.7,
        stock: 10,
        description:
            "Stylish UV protection sunglasses perfect for summer outings. Lightweight frame with polarized lenses that block 99% of UV rays. Perfect for beach days and outdoor adventures.",
        image: "",
        category: "Accessories",
        emoji: "🕶️",
    },
    {
        id: 2,
        name: "Tropical Print Shirt",
        brand: "BeachWear",
        price: 29,
        rating: 4.5,
        stock: 15,
        description:
            "Breathable cotton shirt with vibrant tropical prints. Relaxed fit that keeps you cool during hot summer days. Machine washable and fade-resistant.",
        image: "",
        category: "Clothing",
        emoji: "🌴",
    },
    {
        id: 3,
        name: "SPF 50 Sunscreen",
        brand: "SkinGuard",
        price: 12,
        rating: 4.9,
        stock: 25,
        description:
            "Broad-spectrum SPF 50 sunscreen with water-resistant formula. Lightweight, non-greasy texture that absorbs quickly. Protects against UVA and UVB rays for up to 80 minutes.",
        image: "",
        category: "Skincare",
        emoji: "🧴",
    },
    {
        id: 4,
        name: "Beach Tote Bag",
        brand: "CoastalCo",
        price: 22,
        rating: 4.3,
        stock: 8,
        description:
            "Spacious beach tote with waterproof lining and multiple pockets. Sand-resistant fabric that's easy to clean. Fits towels, snacks, and all your beach essentials.",
        image: "",
        category: "Accessories",
        emoji: "👜",
    },
    {
        id: 5,
        name: "Cooling Mist Spray",
        brand: "FreshBreeze",
        price: 9,
        rating: 4.6,
        stock: 30,
        description:
            "Instant cooling facial mist with aloe vera and cucumber extract. Refreshes and hydrates skin on hot days. Travel-friendly size perfect for on-the-go relief.",
        image: "",
        category: "Skincare",
        emoji: "💨",
    },
    {
        id: 6,
        name: "Flip Flops Deluxe",
        brand: "WaveStep",
        price: 18,
        rating: 4.4,
        stock: 20,
        description:
            "Premium memory foam flip flops with arch support. Non-slip sole perfect for poolside and beach walks. Available in multiple vibrant summer colors.",
        image: "",
        category: "Footwear",
        emoji: "🩴",
    },
];


function Home() {
    const heroTitle =
        window.elementSdk?.config?.hero_title || "Summer Sale 50% OFF";
    const featured = products.slice(0, 3);
    return (
        <>
            <section className="relative overflow-hidden bg-gradient-to-br from-sun-100 via-sun-50 to-ocean-50 px-4 md:px-8 py-16 md:py-24">
                <div className="absolute top-10 right-10 text-6xl md:text-8xl opacity-20 float-anim">☀️</div>
                <div className="absolute bottom-10 left-10 text-4xl md:text-6xl opacity-15 float-anim" style={{ animationDelay: "1s" }}>🌊</div>
                <div className="max-w-7xl mx-auto text-center relative z-10">
                    <div className="inline-block px-4 py-1.5 bg-sun-200/60 rounded-full text-sun-700 text-xs font-semibold mb-4 tracking-wide">🔥 HOT DEALS THIS SEASON</div>
                    <h1 id="hero-title" className="text-4xl md:text-6xl lg:text-7xl font-extrabold leading-tight mb-4 bg-gradient-to-r from-sun-600 via-orange-500 to-ocean-600 bg-clip-text text-transparent">{heroTitle}</h1>
                    <p className="text-gray-600 text-lg md:text-xl max-w-2xl mx-auto mb-8">Discover premium summer essentials at unbeatable prices. Limited time offers on sunglasses, skincare & more.</p>
                    <button className="px-8 py-3.5 bg-gradient-to-r from-sun-500 to-sun-600 text-white font-semibold rounded-2xl shadow-lg hover:shadow-xl hover:scale-105 transition-all pulse-glow">Shop Now <i data-lucide="arrow-right" className="w-4 h-4 inline ml-1"></i></button>
                </div>
            </section>
            <section className="px-4 md:px-8 py-16 max-w-7xl mx-auto">
                <div className="flex items-center justify-between mb-8">
                    <h2 className="text-2xl md:text-3xl font-bold">Popular Products</h2>
                    <a href="#" className="text-sun-600 font-medium text-sm hover:underline">View All →</a>
                </div>
                <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6">
                    {
                        featured.map((p, i) => (
                            <ProductCard key={i} p={p} i={i} />
                        ))
                    }
                </div>
            </section>



            <section className="px-4 md:px-8 py-16 bg-gradient-to-r from-ocean-50 to-sun-50">
                <div className="max-w-7xl mx-auto">
                    <h2 className="text-2xl md:text-3xl font-bold text-center mb-10">
                        Summer Care Tips
                    </h2>

                    <div className="grid grid-cols-1 md:grid-cols-3 gap-6">
                        {[
                            {
                                icon: "droplets",
                                title: "Stay Hydrated",
                                desc: "Drink at least 8 glasses of water daily. Add lemon or cucumber for a refreshing twist.",
                            },
                            {
                                icon: "shield",
                                title: "Sun Protection",
                                desc: "Apply SPF 30+ sunscreen every 2 hours. Wear hats and UV-protective clothing.",
                            },
                            {
                                icon: "heart",
                                title: "Skin Care",
                                desc: "Moisturize after sun exposure. Use aloe vera gel to soothe sun-kissed skin.",
                            },
                        ].map((t, index) => (
                            <div
                                key={index}
                                className="bg-white/80 rounded-2xl p-6 border border-gray-100 card-hover"
                            >
                                <div className="w-12 h-12 bg-ocean-100 rounded-xl flex items-center justify-center mb-4">
                                    <i
                                        data-lucide={t.icon}
                                        className="w-6 h-6 text-ocean-600"
                                    ></i>
                                </div>

                                <h3 className="font-semibold text-lg mb-2">
                                    {t.title}
                                </h3>

                                <p className="text-gray-500 text-sm leading-relaxed">
                                    {t.desc}
                                </p>
                            </div>
                        ))}
                    </div>
                </div>
            </section>


        </>
    )
}

export default Home
